lib/sinon/util/core/next-tick.js

```diff
@@ -0,0 +1,20 @@
+"use strict";
+
+var hasNextTick = typeof process === "object" && typeof process.nextTick === "function";
+var hasSetImmediate = typeof setImmediate === "function";
+
+function nextTick(callback) {
+    setTimeout(callback, 0);
+}
+
+if (hasNextTick) {
+    module.exports = process.nextTick;
+}
+
+if (!hasNextTick && hasSetImmediate) {
```

Comment:
in terms of async scheduling, `nextTick` and `setImmediate` aren't exactly the same - `nextTick` will append the call to the current spin of the loop, and `setImmediate` schedules it for the next, whenever that happens. Maybe `Promise.resolve().then(callback)` would be closer, as that will (afaik) do the same thing as `nextTick`...? It's subtle, but it might be closer to the intended semantics.

ie:

```
function ping(m){
    return function(){
        console.log(m)
    }
}

ping('start')()
setImmediate(ping('immediate'))
ping('start2')()
setTimeout(ping('timeout'),0)
ping('start3')()
process.nextTick(ping('tick'))
ping('start4')()
Promise.resolve().then(ping('promise'))
ping('start5')()


```

gives:
```
start
start2
start3
start4
start5
tick
promise
immediate
timeout
```

Reply:
it'll be slightly faster when making lots of calls too.